assets/js/components/user-input/UserInputQuestionnaire.js

```diff
@@ -125,6 +126,15 @@ export default function UserInputQuestionnaire() {
 		setActiveSlug( steps[ steps.length - 1 ] );
 	}, [ activeSlugIndex ] );
 
+	useEffect( () => {
+		if ( firstRender ) {
+			setFirstRender( false );
+			return;
+		}
```

Comment:
Alternatively, we could maybe just move the effect here to its cleanup function so that it's only invoked on the "backside" which would also skip the initial run like we want. As long as that didn't trigger an unwanted scroll at the very end, I would think this might be the better way to go. What do you think?

Reply:
I think that's a bit more confusing—it's clever, but it's not straightforward.

As a rule I'd expect to look in a `useEffect`'s main function for its intended effects, and its returned function for the cleanup. In that case, we're be running the intended effect in the cleanup. It would work, but I think it's a bit too clever for its own good 😅